File: Gravels Extended Battles 1.21/common/src/main/java/drai/dev/gravelsextendedbattles/requirements/FriendshipBelowRequirement.java
package drai.dev.gravelsextendedbattles.requirements;

import com.cobblemon.mod.common.api.pokemon.requirement.*;
import com.cobblemon.mod.common.pokemon.*;
import org.jetbrains.annotations.*;

public class FriendshipBelowRequirement implements Requirement {
    int amount = 0;
    public static String ADAPTER_VARIANT = "friendship_below";
    
    @Override
    public boolean check(@NotNull Pokemon pokemon) {
        return pokemon.getFriendship() <= this.amount;
    }
}

File: common/src/main/kotlin/drai/dev/gravelsextendedbattles/additions/evolutions/FriendshipBelowRequirement.kt
package drai.dev.gravelsextendedbattles.additions.evolutions

import com.cobblemon.mod.common.api.pokemon.requirement.Requirement
import com.cobblemon.mod.common.pokemon.Pokemon

class FriendshipBelowRequirement : Requirement {
    var amount: Int = 0
    override fun check(pokemon: Pokemon): Boolean {
        return pokemon.friendship <= this.amount
    }

    companion object {
        var ADAPTER_VARIANT: String = "friendship_below"
    }
}
